src/Microsoft.AspNet.JsonPatch/Helpers/ExpandoObjectDictionaryExtensions.cs

```diff
@@ -0,0 +1,72 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.JsonPatch.Helpers
+{
+    // Helper methods to allow case-insensitive key search
+    internal static class ExpandoObjectDictionaryExtensions
+    {
+        internal static void SetValueForCaseInsensitiveKey(this IDictionary<string, object> propertyDictionary,
```

Comment:
;-)  tbh, when I started working on this I figured I'd be able to create it with a case-insensitive comp, but: no such luck.  Yet looking deeper into it, it makes sense - keys should be case-sensitive for ExpObj, as they relate to .NET properties - "name != Name".  

I wrote these extensions as they're used throughout the analysisresult/objectadapter; these methods keep the code in one place.  If you prefer, I could use helper methods rather than extension methods?